dask/array/tests/test_masked.py

```diff
@@ -379,7 +380,11 @@ def test_average_weights_with_masked_array():
     d_weights = da.from_array(weights, chunks=2)
 
     np_avg = np.ma.average(a, weights=weights, axis=1)
-    da_avg = da.ma.average(d_a, weights=d_weights, axis=1)
+    if keepdims:
+        # np.average only takes keepdims keyword from v1.23, so use reshape to
+        # form the expected array.
+        np_avg = np_avg.reshape((3, 1))
```

Comment:
I assume you test against older numpy versions, otherwise these could obviously be simpler.